src/python/WMCore/MicroService/MSRuleCleaner/MSRuleCleaner.py

```diff
@@ -545,12 +597,18 @@ def archive(self, wflow):
         # Make all the needed checks before trying to archive
         if not (wflow['IsClean'] or wflow['ForceArchive']):
             msg = "Not properly cleaned workflow: %s" % wflow['RequestName']
+            if self._isStatusAdvanceExpired(wflow, wflow['RequestStatus']):
+                self.alertStatusAdvanceExpired(wflow, msg)
```

Comment:
Can't we have this check being done only once in the method upstream (after catching these exceptions)?

Reply:
I was thinking the same at the beginning, but then we loose the additional info provided by the message. Which contains really important information about why the workflow is sitting stuck and not progressing, Meaning at which point it was ejected out of the pipeline.

Reply:
You raise an exception and you provide the error message. So you could catch it upstream and make it part of the error message (as we are discussing in a different comment).